modelator/src/event.rs

```diff
@@ -45,11 +45,11 @@ pub enum Event {
     /// Process the abstract action, modifying the system state.
     Action(Box<dyn Any>),
     /// Expect the provided outcome of the last action.
-    Outcome(String),
+    Expect(String),
     /// Check the assertion about the abstract system state.
     Check(Box<dyn Any>),
     /// Expect exactly the provided abstract system state.
-    Expect(Box<dyn Any>),
+    Equal(Box<dyn Any>),
```

Comment:
IIUC, this `Event::Equal` is the reason why we need `StateHandler::read` right?

I'm thinking that implementing `StateHandler::read` can be a lot of effort (and some users may not even need it). Not only users have to provide a way to go from an abstract state to a concrete one (the bare minimum), now they also need to provide the opposite transformation. Moreover, now users have to implement two traits, `StateHandler` and `ActionHandler`.

Overall, I would be in favor of simplifying things (probably because I don't understand why we had to complicate them in the first place): I would have a single trait that only has two methods, `init` and `handle`. This would already provide basic functionality. Then, we should also support more complicated setups for which this is not enough, but not at the cost of complicating the rest.

Reply:
I just realized that what advocating for is very similar to the initial approach we had in the `TestRunner`: https://github.com/informalsystems/modelator/blob/e3e859d70694d5c79ddd414f8668f4506157e608/modelator/src/runner.rs#L8-L14